packagesFromOverlaysBuilderConstructor.nix

```diff
@@ -1,89 +1,76 @@
 { flake-utils-plus }:
 let
 
-  overlayFromPackagesBuilderConstructor = channels:
+  overlayFromPackagesBuilderConstructor = overlays:
     let
-      # channels: channels.<name>.overlays
+      # overlays: self.overlays
 
-      overlayFromPackagesBuilder = pkgs:
+      overlayFromPackagesBuilder = channels:
         /**
-          Synopsis: overlayFromPackagesBuilder _pkgs_
+          Synopsis: overlayFromPackagesBuilder _channels_
 
-          pkgs:     pkgs.<system>.<tree>
+          channels: builder `channels` argument
 
           Returns valid packges that have been defined within an overlay so they
           can be shared via _self.packages_ with the world. This is especially useful
           over sharing one's art via _self.overlays_ in case you have a binary cache
           running from which third parties could benefit.
 
-          First, flattens an arbitrarily nested _pkgs_ tree for each system into a flat
-          key in which nesting is aproximated by a "/" (e.g. "development/kakoune").
-          Also filter the resulting packages for attributes that _trivially_ would
-          fail flake checks (broken, system not supported or not a derivation).
+          Steps:
+           1. merge all channels into one nixpkgs attribute set
+           2. collect all overlays' packages' keys into one flat list
+           3. pick out each package from the nixpkgs set into one packages set
+           $. flatten package set and filter out disallowed packages - by flake check requirements
 
-          Second, collects all overlays' packages' keys of all channels into a flat list.
+          example input and output:
+            overlays = {
+              "unstable/firefox" = prev: final: {
+                firefox = prev.override { privacySupport = true; };
+              }; 
+            }
 
-          Finally, only passes packages through the seive that are prefixed with a top level
-          key exposed by any of the overlays. Since overlays override (and do not emrge)
-          top level attributes, by filtering on the prefix, an overlay's entire packages
-          tree will be correctly captured.
-
-          Example:
-
-          pkgs'.<system> = {
-          "development/kakoune" = { ... };
-          "development/vim" = { ... };
-          };
-
-          overlays' = [
-          "development"
-          ];
-
-          overlays' will pass both pkgs' through the sieve.
+            self.packages = {
+              firefox = prev.override { privacySupport = true; };
```

Comment:
I reckon technically, it couldn't be `prev.`, I think you wanted to put in a placeholder that represents a `<derivation-with-overlay-applied>`

Reply:
good point, just wrote out what I was going for in text